be/src/storage/lake/schema_change.cpp

```diff
@@ -328,6 +328,60 @@ Status SchemaChangeHandler::do_process_alter_tablet(const TAlterTabletReqV2& req
     return Status::OK();
 }
 
+Status SchemaChangeHandler::process_update_tablet_meta(const TUpdateTabletMetaInfoReq& request) {
+    if (!request.__isset.txn_id) {
+        return Status::InternalError("txn_id not be set");
+    }
+    int64_t txn_id = request.txn_id;
+
+    for (const auto& tablet_meta_info : request.tabletMetaInfos) {
+        auto status = do_process_update_tablet_meta(tablet_meta_info, txn_id);
+        if (!status.ok()) {
+            return status;
+        }
+    }
+
+    return Status::OK();
+}
+
+Status SchemaChangeHandler ::do_process_update_tablet_meta(const TTabletMetaInfo& tablet_meta_info, int64_t txn_id) {
+    MonotonicStopWatch timer;
+    timer.start();
+    LOG(INFO) << "begin to update tablet. tablet: " << tablet_meta_info.tablet_id
+              << "update meta type:" << tablet_meta_info.meta_type;
+
+    auto tablet_id = tablet_meta_info.tablet_id;
+    ASSIGN_OR_RETURN(auto tablet, _tablet_manager->get_tablet(tablet_id));
+
+    if (tablet.get_txn_log(txn_id).ok()) {
+        LOG(INFO) << "the update tablet meta task has been processed, txn_id:" << txn_id;
+        return Status::OK();
+    }
+
+    // create txn log
+    auto txn_log = std::make_shared<TxnLog>();
+    txn_log->set_tablet_id(tablet_id);
+    txn_log->set_txn_id(txn_id);
+    auto op_alter_meta = txn_log->mutable_op_alter_meta();
+
+    if (tablet_meta_info.meta_type != TTabletMetaType::ENABLE_PERSISTENT_INDEX) {
```

Comment:
This check should be done at the beginning of the function.

Reply:
ok